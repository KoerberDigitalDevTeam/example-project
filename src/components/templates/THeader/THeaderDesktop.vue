<template>
  <div class="theader-desktop__container">
    <h3 v-testId.theader-desktop.title class="theader-desktop__title">{{ title }}</h3>
  </div>
</template>

<script>
export default {
  name: 'THeaderDesktop',

  props: {
    title: {
      type: String,
      required: true,
    },
  },
};
</script>

<style lang="scss" scoped>
.theader-desktop__container {
  padding: 16px 0 16px 24px;
}
</style>
