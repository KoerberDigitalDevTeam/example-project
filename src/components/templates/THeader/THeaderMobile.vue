<template>
  <div class="theader-mobile--sticky">
    <div class="theader-mobile">
      <AButton
        v-testId.theader.mobile-button
        :tertiary="true"
        icon
        color="dWhite"
        class="theader-mobile__button mr-6"
        data-testid="btn-mob-header-open"
        @click.stop="handleToggleSidebar"
      >
        <v-icon>mdi-menu</v-icon>
      </AButton>

      <v-list-item-title v-testId.theader.mobile-title class="theader-mobile__title">
        <h4>{{ title }}</h4>
      </v-list-item-title>
    </div>
  </div>
</template>

<script>
import { mapState, mapActions } from 'vuex';
import AButton from '@/components/atoms/AButton/AButton.vue';

export default {
  name: 'THeaderMobile',
  components: {
    AButton,
  },

  props: {
    title: {
      type: String,
      required: true,
    },
  },

  data() {
    return {
      isSidebarOpen: false,
    };
  },

  computed: {
    ...mapState({
      isSidebarOpenGlobal: (state) => state.global.isSidebarOpen,
    }),
  },

  watch: {
    isSidebarOpenGlobal(newValue) {
      this.isSidebarOpen = newValue;
    },
  },

  methods: {
    ...mapActions({ toggleSidebar: 'global/toggleSidebar' }),
    handleToggleSidebar() {
      this.isSidebarOpen = !this.isSidebarOpen;
      this.toggleSidebar(this.isSidebarOpen);
    },
  },
};
</script>

<style lang="scss" scoped>
.theader-mobile {
  display: flex;
  background: $d-black;
  color: $d-white;
  padding: 4px 0 4px 20px;
  overflow: hidden;
}

.theader-mobile--sticky {
  position: sticky;
  top: 0;
  z-index: 4;
}
</style>
