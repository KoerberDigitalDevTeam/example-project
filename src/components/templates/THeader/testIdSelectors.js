export default {
  title: 'header-title',
  toolbar: 'header-toolbar',
  toolbarLeft: 'header-toolbar-left',
  toolbarRight: 'header-toolbar-right',
};
