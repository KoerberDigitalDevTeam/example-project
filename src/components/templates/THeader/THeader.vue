<template>
  <component
    :is="['THeaderDesktop', 'THeaderMobile'][+$vuetify.breakpoint.mobile]"
    v-bind="{ ...$props, ...$attrs }"
    v-on="$listeners"
  />
</template>

<script>
import THeaderDesktop from './THeaderDesktop.vue';
import THeaderMobile from './THeaderMobile.vue';

export default {
  name: 'THeader',
  components: {
    THeaderDesktop,
    THeaderMobile,
  },
};
</script>
