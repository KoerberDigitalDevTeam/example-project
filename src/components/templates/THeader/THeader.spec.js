import { shallowMount } from '@vue/test-utils';
import Vuetify from 'vuetify';
import THeader from '@/components/templates/THeader/THeader.vue';

function createWrapper(options = {}) {
  return shallowMount(THeader, {
    vuetify: new Vuetify(),
    ...options,
  });
}

describe('THeader', () => {
  describe('Snapshot', () => {
    it('Should THeader keep structure', () => {
      const wrapper = createWrapper({
        propsData: {
          title: 'Test title',
        },
      });

      expect(wrapper.html()).toMatchSnapshot();
    });
  });
});
