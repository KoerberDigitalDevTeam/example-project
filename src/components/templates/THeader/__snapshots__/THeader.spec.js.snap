// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`THeader Snapshot Should THeader keep structure 1`] = `<theadermobile-stub title="Test title"></theadermobile-stub>`;
